This patch fixes the build of Libreoffice 6.0.5 with icu 61.1.
It was downloaded from https://bugs.gentoo.org/651702.

--- a/forms/source/xforms/datatypes.hxx	
+++ a/forms/source/xforms/datatypes.hxx	
@@ -36,6 +36,8 @@ 
 
 #include <memory>
 
+using icu::RegexMatcher;
+
 
 namespace xforms
 {
--- a/i18npool/inc/collator_unicode.hxx	
+++ a/i18npool/inc/collator_unicode.hxx	
@@ -26,6 +26,8 @@ 
 
 #include <unicode/tblcoll.h>
 
+using icu::RuleBasedCollator;
+
 //      ----------------------------------------------------
 //      class Collator_Unicode
 //      ----------------------------------------------------
--- a/i18npool/source/breakiterator/breakiterator_unicode.cxx	
+++ a/i18npool/source/breakiterator/breakiterator_unicode.cxx	
@@ -30,6 +30,9 @@ 
 #include <rtl/ustring.hxx>
 #include <string.h>
 
+using icu::BreakIterator;
+using icu::RuleBasedBreakIterator;
+
 U_CDECL_BEGIN
 extern const char OpenOffice_dat[];
 U_CDECL_END
--- a/i18npool/source/collator/collator_unicode.cxx	
+++ a/i18npool/source/collator/collator_unicode.cxx	
@@ -29,6 +29,8 @@ 
 #include <com/sun/star/i18n/CollatorOptions.hpp>
 #include <cppuhelper/supportsservice.hxx>
 
+using icu::Collator;
+
 using namespace ::com::sun::star;
 using namespace ::com::sun::star::i18n;
 using namespace ::com::sun::star::lang;
--- a/i18npool/source/collator/gencoll_rule.cxx	
+++ a/i18npool/source/collator/gencoll_rule.cxx	
@@ -31,6 +31,8 @@ 
 
 #include <unicode/tblcoll.h>
 
+using icu::RuleBasedCollator;
+
 /* Main Procedure */
 
 void data_write(char* file, char* name, sal_uInt8 *data, sal_Int32 len)
--- a/i18npool/source/ordinalsuffix/ordinalsuffix.cxx	
+++ a/i18npool/source/ordinalsuffix/ordinalsuffix.cxx	
@@ -29,6 +29,8 @@ 
 #include <unicode/normlzr.h>
 #include <memory>
 
+using icu::NumberFormat;
+
 using namespace ::com::sun::star::i18n;
 using namespace ::com::sun::star::uno;
 using namespace ::com::sun::star;
--- a/i18npool/source/transliteration/ignoreDiacritics_CTL.cxx	
+++ a/i18npool/source/transliteration/ignoreDiacritics_CTL.cxx	
@@ -12,6 +12,8 @@ 
 #include <unicode/translit.h>
 #include <unicode/uchar.h>
 
+using icu::UnicodeString;
+
 namespace i18npool {
 
 ignoreDiacritics_CTL::ignoreDiacritics_CTL()
--- a/i18nutil/source/utility/unicode.cxx	
+++ a/i18nutil/source/utility/unicode.cxx	
@@ -34,6 +34,9 @@ 
 // which (obviously) breaks UnicodeType::CURRENCY_SYMBOL
 #undef CURRENCY_SYMBOL
 
+using icu::NumberFormat;
+using icu::UnicodeString;
+
 using namespace ::com::sun::star::i18n;
 
 template<class L, typename T>
--- a/lotuswordpro/source/filter/localtime.cxx	
+++ a/lotuswordpro/source/filter/localtime.cxx	
@@ -57,6 +57,8 @@ 
 #include <limits.h>
 #include <unicode/timezone.h>
 
+using icu::TimeZone;
+
 const long DAY_SEC =24 * 60 * 60;
 const long YEAR_SEC = 365 * DAY_SEC;
 const long FOURYEAR_SEC = 4 * YEAR_SEC + DAY_SEC;
--- a/opencl/source/openclconfig.cxx	
+++ a/opencl/source/openclconfig.cxx	
@@ -17,6 +17,8 @@ 
 #include <sal/log.hxx>
 #include <sal/types.h>
 
+using icu::RegexMatcher;
+
 OpenCLConfig::OpenCLConfig() :
     mbUseOpenCL(true)
 {
--- a/vcl/inc/scrptrun.h	
+++ a/vcl/inc/scrptrun.h	
@@ -45,6 +45,8 @@ 
 #include <unicode/uscript.h>
 #include <vector>
 
+using icu::UObject;
+
 namespace vcl {
 
 struct ParenStackEntry
